rosbag2_py/src/rosbag2_py/_transport.cpp

```diff
@@ -121,8 +123,19 @@ class Player
       }
     }
 
-    rosbag2_transport::Player impl(std::move(reader));
-    impl.play(storage_options, play_options);
+    auto player = std::make_shared<rosbag2_transport::Player>(
+      std::move(reader), storage_options, play_options);
+
+    rclcpp::executors::SingleThreadedExecutor exec;
+    exec.add_node(player);
+    auto spin_thread = std::thread(
+      [&exec]() {
+        exec.spin();
+      });
+    player->play();
```

Comment:
it seems that this does not handle `play_options.loop` anymore - that loop was in `rosbag2_transport.cpp` - or am I missing its new location?

Reply:
Oh - my mistake, yes I remember now that you added it to `Player` in the previous PR